lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -22,15 +22,19 @@
 #include "triton/Dialect/TritonGPU/Transforms/Utility.h"
 #include <memory>
 
-#define GEN_PASS_CLASSES
+namespace mlir::triton::gpu {
+#define GEN_PASS_DEF_TRITONGPUREMOVELAYOUTCONVERSIONS
 #include "triton/Dialect/TritonGPU/Transforms/Passes.h.inc"
+} // namespace mlir::triton::gpu
```

Comment:
How about put this whole file inside namespace mlir::triton::gpu::<anonymous>?  Then we don't need the `using` decls and we don't need to spell out `mlir::triton::gpu::impl::TritonGPURemoveLayoutConversionsBase`

Reply:
Same in other files

Reply:
The majority of the passes used the approach with `using` declarations before, so I opted to change `RemoveLayoutConversions` in that direction. to align it with the other passes while having a small diff.

However, if you prefer to use the approach where we put the whole file inside the namespace `mlir::triton::gpu`, I can make that change to the files in this PR.

Reply:
I think a problem with the way it was before is that all "using namespace foo" decls have the same priority, so if there's a class mlir::Foo and mlir::triton::Foo and you type `Foo`, that's ambiguous.  But if you're inside namespace mlir::triton, then plain `Foo` is not ambiguous.  https://gcc.godbolt.org/z/9Eq3Pr1c8

IOW if we do the three-`using` approach, then we almost might as well put all of Triton into namespace `mlir`.  So I think we should move away from that.

Reply:
Makes sense, thanks!

I've updated this PR to put the pass implementations inside the `mlir::triton::gpu` namespace and remove the `using` declarations.